app_flutter/lib/widgets/task_grid.dart

```diff
@@ -47,12 +49,367 @@ class TaskGridContainer extends StatelessWidget {
+        showLuci == other.showLuci
+    ;
+  }
+}
+
+class TaskGridFilterWidget extends StatefulWidget {
+  const TaskGridFilterWidget(this.filterNotifier, this.onClose);
+
+  final ValueNotifier<TaskGridFilter> filterNotifier;
+  final Function() onClose;
+
+  @override
+  State createState() => TaskGridFilterState();
+}
+
+class TaskGridFilterState extends State<TaskGridFilterWidget> {
+  @override
+  void initState() {
+    super.initState();
+    widget.filterNotifier.addListener(_update);
+    taskController = TextEditingController(text: filter.taskFilter?.pattern ?? '');
+    authorController = TextEditingController(text: filter.authorFilter?.pattern ?? '');
+    messageController = TextEditingController(text: filter.messageFilter?.pattern ?? '');
+  }
+
+  void _update() {
+    setState(() {});
+  }
+
+  @override
+  void dispose() {
+    widget.filterNotifier.removeListener(_update);
+    super.dispose();
+  }
+
+  static TextStyle labelStyle = const TextStyle(
+    color: Colors.black,
+    fontSize: 16.0,
+    fontWeight: FontWeight.normal,
+    fontStyle: FontStyle.normal,
+    decoration: TextDecoration.none,
+  );
+
+  TaskGridFilter get filter => widget.filterNotifier.value;
+  TextEditingController taskController;
+  TextEditingController authorController;
+  TextEditingController messageController;
+
+  void _setFilter(TaskGridFilter newFilter) {
+    widget.filterNotifier.value = newFilter;
+  }
+
+  void _newTaskFilter(RegExp newRegExp) {
+    _setFilter(filter.copyWith(taskFilter: newRegExp));
+  }
+
+  void _newAuthorFilter(RegExp newRegExp) {
+    _setFilter(filter.copyWith(authorFilter: newRegExp));
+  }
+
+  void _newMessageFilter(RegExp newRegExp) {
+    _setFilter(filter.copyWith(messageFilter: newRegExp));
+  }
+
+  void _newShowAndroid(bool newValue) {
+    _setFilter(filter.copyWith(showAndroid: newValue));
+  }
+
+  void _newShowIos(bool newValue) {
+    _setFilter(filter.copyWith(showIos: newValue));
+  }
+
+  void _newShowWindows(bool newValue) {
+    _setFilter(filter.copyWith(showWindows: newValue));
+  }
+
+  void _newShowCirrus(bool newValue) {
+    _setFilter(filter.copyWith(showCirrus: newValue));
+  }
+
+  void _newShowLuci(bool newValue) {
+    _setFilter(filter.copyWith(showLuci: newValue));
+  }
+
+  Widget _pad(Widget child, Alignment alignment) {
+    return Container(
+      padding: const EdgeInsets.all(5.0),
+      child: child,
+      alignment: alignment,
+    );
+  }
+
+  TableRow _makeRow(String label, Widget editable) {
+    return TableRow(
+      children: <Widget>[
+        _pad(Text(label, style: labelStyle), Alignment.centerRight),
+        _pad(editable, Alignment.centerLeft),
+      ],
+    );
+  }
+
+  TableRow _makeTextFilterRow(String label, TextEditingController controller, void onChanged(RegExp newValue)) {
+    return _makeRow(label,
+      TextField(
+        controller: controller,
+        decoration: const InputDecoration(
+          hintText: '(regular expression)',
```

Comment:
This should be an example regular expression

Reply:
I thought the hints were descriptive rather than exemplary...?

Reply:
My concern is that a user won't know that this dashboard is built in Dart, and it's expecting an ECMAScript formatted regex. So when they try their python regex it fails. Additionally, if they know regex they might not know what advanced options there are.

Can we have the helper/placeholder text be an example and the hint text be something like `ECMAScript/Javascript Regular Expression`?

Reply:
I believe this is resolved in a recent commit.